daemon/routes/iprule/iprule.go

```diff
@@ -80,16 +81,72 @@ func (r *Router) SetupRoutingRules(
 			return err
 		}
 
-		otherRuleID, err := calculateRulePriority(ipv6)
-		if err != nil {
-			return err
+		// PeerA (LAN-a 192.168.1.x) connects to PeerB (LAN-b 192.168.1.x)
+		// if PeerB allows its LAN access when used as Exit node
+		// then PeerB LAN access is the priority over PeerA LAN
+		if !routeToRemoteLan {
+			otherRuleID, err := calculateRulePriority(ipv6)
+			if err != nil {
+				return err
+			}
+
+			if err = addSuppressprefixLengthRule(
+				otherRuleID,
+				ipv6,
+			); err != nil {
+				return err
+			}
 		}
+	}
 
-		if err = addSuppressprefixLengthRule(
-			otherRuleID,
-			ipv6,
-		); err != nil {
-			return err
+	return nil
+}
+
+// PreventLocalTraffic adjust policy based routing rules
+func (r *Router) PreventLocalTraffic(
+	vpnInterface net.Interface,
+	ipv6Enabled bool,
+	routeToRemoteLan bool,
```

Comment:
Continuing with my above suggestion to rename the function, this argument should simply be called `enable`

Reply:
Removed this function, added extra logic to `SetupRoutingRules`